File: dotnet/net6/Watches-v6/IAlarmStrategy.cs
namespace Watches;

public interface IAlarmStrategy
{
    void AddAlarm(Alarm alarm);
    IEnumerable<Alarm> GetAlarms(DateTime? @from = null, DateTime? to = null);
    void RemoveAlarm(Alarm alarm);
    Alarm? FindAlarmByName(string alarmName);
}

File: java/src/com/infoback/IAlarmStrategy.java
package com.infoback;

import java.time.LocalDateTime;

public interface IAlarmStrategy {
  void addAlarm(Alarm alarm);
  Iterable<Alarm> getAlarms(LocalDateTime fromTime, LocalDateTime toTime);
  void removeAlarm(Alarm alarm);
  Alarm findAlarmByName(String alarmName);
}
